pkg/logql/quantile_over_time_sketch.go

```diff
@@ -242,10 +240,14 @@ type quantileSketchBatchRangeVectorIterator struct {
 }
 
 func (r *quantileSketchBatchRangeVectorIterator) At() (int64, StepResult) {
-	if r.at == nil {
-		r.at = make([]ProbabilisticQuantileSample, 0, len(r.window))
-	}
+	/*
+		if r.at == nil {
+			r.at = make([]ProbabilisticQuantileSample, 0, len(r.window))
+		}
+		clear(r.at)
+	*/
 	r.at = r.at[:0]
+	r.at = make([]ProbabilisticQuantileSample, 0, len(r.window))
```

Comment:
I think if we're not reusing `r.at` then reassigning it via this line should be enough to allow the previous underlying array to be gc'd right?